### ISSUE TITLE
NullPointerException: Cannot invoke "org.audiveris.omr.sheet.rhythm.Measure.addInter

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#558]
![Bicycle Race (2)](https://user-images.githubusercontent.com/33522630/155836618-c1a766f0-cb8c-4bcf-9cd8-71e60b1edff0.jpg)

The appended score gives an error message during transcription:
Error in performing [RHYTHMS, PAGE] java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "org.audiveris.omr.sheet.rhythm.Measure.addInter(org.audiveris.omr.sig.inter.Inter)" because "measure" is null.

The reason is unclear: I fixed the error with the missing bar line in the bottom score, but the error stays present.

**Comment by hbitteur** [comment]
Sorry Baruch, I'm very busy right now, I should be back within a week

**Comment by hbitteur** [comment]
@Bacchushlg 
The problems are due to bar-lines correctly recognized but later replaced by false stems. These stems, together with the connected (false) note heads, end up in higher grades than the originally recognized bar-lines, which consequently get discarded.

This partial lack of bar-lines results in inconsistency between measures of the same system from one staff to the other:
- System 1: the two upper staves exhibit 3 measures while the two lower staves have 4 measures
- System 2 is OK
- System 3: its 4th staff lacks 2 bar-lines, thus resulting in just 2 measures while the other staves have 4 measures.

This inconsistency later  leads to IndexOutOfBoundsException (or something similar).

This being said, I succeeded to correctly run the recognition, using a rather subtle approach:
1. I re-ran recognition up until TEXTS step (because it's the last one before MEASURES step)
2. For every discarded bar-line, I manually removed the false stem+head stuff
3. I also manually removed the corresponding bar-lines in the other staves, this is important , because this will lead to false but **consistent** measures across all staves of a system.
4. I then ran the MEASURES step.
5. Finally, I manually inserted the missing bar-lines at the 3 key locations, which resulted in correct measure splits.
Done! Pfff!

Now, there should be more convenient/automatic ways to detect and fix these situations.
An approach might be to manually "freeze" the bar-lines so that they can't get discarded by conflicting symbols.
Another one would be to make Audiveris MEASURES step more strict regarding the alignment of measures in a system.

Stay tuned...

**Comment by hbitteur** [comment]
A simpler way to address the problem manually is the following.
Since the root cause is the conflicting stems, we can:
1. Run until STEMS step
2. Manually remove the conflicting stems (there are 4 stems to remove)
3. Run the following steps

**Comment by hbitteur** [comment]
There was indeed a bug in the measure consistency enforcement code, which resulted in no effective control. Now fixed by commit 18fdfcd2c4f7bf6acce2d4363a4e5edad7b3e57a

Here are the messages printed by MEASURES step on this score example:
```
[558] S1 partial barline column: s3:G(x:1862,y:815)[#34] s4:G(x:1863,y:1036)[#46]
[558] S1 deleting   half column: s3:G(x:1862,y:815)[#34] s4:G(x:1863,y:1036)[#46]
[558] S3 partial barline column: s9:G(x:804,y:2458)[#102] s10:G(x:805,y:2682)[#116] s11:G(x:805,y:2899)[#130]
[558] S3 expanding major column: s9:G(x:804,y:2458)[#102] s10:G(x:805,y:2682)[#116] s11:G(x:805,y:2899)[#130]
[558] S3 partial barline column: s9:G(x:1312,y:2456)[#104] s10:G(x:1313,y:2681)[#118] s11:G(x:1313,y:2897)[#132]
[558] S3 expanding major column: s9:G(x:1312,y:2456)[#104] s10:G(x:1313,y:2681)[#118] s11:G(x:1313,y:2897)[#132]
[558] 11 raw measures: [3 in system#1, 4 in system#2, 4 in system#3]
[558] End of sheet step MEASURES
```

The algorithm detected and fixed 3 column problems (one on system 1 and two on system 3):
- S1 at x1862: 2 staves without bar-line vs 2 staves with bar-line, no majority found, so the bar-line column is simply deleted
- S3 at x804: 1 staff without bar-line vs 3 staves with bar-line, majority is to expand the bar-line
- S3 at x1312: 1 staff without bar-line vs 3 staves with bar-line, majority is to expand the bar-line

The two last decisions are correct, the first one is not (because there was no clear majority) and will need a manual measure insertion.

**Comment by Bacchushlg** [comment]
it's working perfectly now
